app/src/main/java/org/projectbuendia/client/utils/Utils.java

```diff
@@ -79,36 +84,65 @@
                 return ((String) a).compareTo((String) b);
             }
             return (a == null ? 0 : intA != null ? 1 : 2)
-                - (b == null ? 0 : intB != null ? 1 : 2);
+                    - (b == null ? 0 : intB != null ? 1 : 2);
         }
     };
     /**
      * Compares two lists, each of whose elements is a null, Integer, Long,
      * BigInteger, or String, lexicographically by element, just like Python.
      */
     public static Comparator<List<Object>> nullIntStrListComparator =
-        new Comparator<List<Object>>() {
-            @Override public int compare(List<Object> a, List<Object> b) {
-                for (int i = 0; i < Math.min(a.size(), b.size()); i++) {
-                    int result = nullIntStrComparator.compare(a.get(i), b.get(i));
-                    if (result != 0) {
-                        return result;
+            new Comparator<List<Object>>() {
+                @Override
+                public int compare(List<Object> a, List<Object> b) {
+                    for (int i = 0; i < Math.min(a.size(), b.size()); i++) {
+                        int result = nullIntStrComparator.compare(a.get(i), b.get(i));
+                        if (result != 0) {
+                            return result;
+                        }
                     }
+                    return a.size() - b.size();
                 }
-                return a.size() - b.size();
-            }
-        };
+            };
+
+
+    /**
+     * Sorts an ArrayList of ObsRow rows so that the order of the ObsRow.conceptUids matches the
+     * order of conceptRowUuids.
+     */
+    public static ArrayList<ObsRow> sortObsRows(final ArrayList<ObsRow> rows,
+                                                final ArrayList<String> conceptUuids) {
+
+        Collections.sort(rows, new Comparator<ObsRow>() {
+                    @Override
+                    public int compare(ObsRow obsRow1, ObsRow obsRow2) {
+                        int indexObsRow1 = conceptUuids.indexOf(obsRow1.conceptUid);
+                        int indexObsRow2 = conceptUuids.indexOf(obsRow2.conceptUid);
+                        if (indexObsRow1 == -1) {
+                            indexObsRow1 += rows.size();
```

Comment:
It looks like this means any row that doesn't appear in the `conceptUuids` list will get an index of `rows.size() - 1`.  I see two possible problems with this:
  - If there is one row for every concept, then `rows.size() - 1` is also the index of the last row in the list, which would mean that it becomes indistinguishable (in the ordering) from the last item, and might end up sorted before or after the last item.  Or basically, the `+=` isn't needed because we don't care about the `-1`; we just want it to be something bigger than any other matching row, for example, set it to `conceptUuids.size()`.
  - The non-matching items all end up with the same index, so they will get sorted into a random-looking order.  There isn't really a great way to standardize on an order to put them in, but let's make it predictable, at least—we could do a secondary sort by comparing the concept names when the indexes are equal.

Reply:
Would you prefer a secondary sort by name? Or maintaining the order in the input list?

Reply:
Secondary sort by name would be better—I don't think the order in the input list is useful.